src/shapes/IText/ITextBehavior.ts

```diff
@@ -805,9 +805,10 @@ export abstract class ITextBehavior<
     copiedStyle?: { [index: number]: TextStyleDeclaration }
   ) {
     const newLineStyles: { [index: number]: TextStyleDeclaration } = {};
-    const isEndOfLine =
-      this._unwrappedTextLines[lineIndex].length === charIndex;
-    let somethingAdded = false;
+    const originalLineLength = this._unwrappedTextLines[lineIndex].length;
+    const isEndOfLine = originalLineLength === charIndex;
+
+    let someStyleIsCarringOver = false;
```

Comment:
Do you mean someStyleIsCarryingOver?

Reply:
@gloriousjob you can use the suggestion markdown
![image](https://github.com/fabricjs/fabric.js/assets/34343793/14198132-bfe7-481f-a8d8-ca8a8b65ed60)

```suggestion
    let someStyleIsCarryingOver = false;
```